### DISCUSSION TITLE
Problem Query Middleware run many times

### DISCUSSION THREAD

**Discussion by IRediTOTO** [prisma/prisma#8988] [Q&A]
Hi, I am using NextJS. 

I have global Prisma
```ts
let prm: PrismaClient

if (process.env.NODE_ENV === 'production' && typeof window === 'undefined') {
    prm = new PrismaClient()
    prm.$use(transformUpdateAt)
} else {
    if (!global.prisma && typeof window === 'undefined') {
        global.prisma = new PrismaClient()
    }
    global.prisma.$use(transformUpdateAt)

    prm = global.prisma
}

export default prm
```
the middleware detail is:
```ts
export const transformUpdateAt: Prisma.Middleware = async (params, next) => {
    if (params.model === 'Products') {
        const res = await next(params)
        if (res?.updateAt) {
            console.log('res?.updateAt', res?.updateAt)
            res.updateAt = moment(res.updateAt).format('MM/DD/yyyy')
        }
        return res
    }
    return await next(params)
}
```
When I ran call api `/api/test` the console log show 3 times 
```
res?.updateAt 2021-07-23T07:35:06.772Z
res?.updateAt 07/23/2021
res?.updateAt 07/23/2021
res?.updateAt 07/23/2021
res?.updateAt 07/23/2021
```
Can you tell me what is happening here ?

**Comment by IRediTOTO** [comment]
The query grew time by time. Look like I got wrong config :( 
![image](https://user-images.githubusercontent.com/15695252/131250769-fdc0c904-1e38-4625-88fe-23a851a757da.png)

**Comment by ryands17** [comment]
The logs don't suggest how many times the query has run. To know the query run, you need to add this:

```ts
global.prisma = new PrismaClient({
  log: ['query']
})
```

This will log the generated SQL and then you can check how many times it was called.

**Reply by IRediTOTO** [reply]
I console log in `Prisma.Middleware` function. I think because I am using `npm run dev`.

**Reply by IRediTOTO** [reply]
Now the problem is just about 100% CPU database. My db is small and simple, I don't think it can get 100% CPU. And I just run local now.

**Comment by janpio** [comment]
You are adding another instance of the middleware each time, although the `global.prisma` is reused. You need to move the `$use` call into that `if` as well. Otherwise although it recognizes that you already have a Prisma and reuses that, it will still add another instance of the middleware.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
